workflow-ui/container-android/src/main/java/com/squareup/workflow1/ui/modal/ModalViewContainer.kt

```diff
@@ -71,14 +72,13 @@ public open class ModalViewContainer @JvmOverloads constructor(
     // that should be blocked by this modal session.
     val wrappedRendering = BackButtonScreen(asScreen(initialModalRendering)) { }
 
-    val view = wrappedRendering.buildView(
-      viewEnvironment = initialViewEnvironment,
+    val viewHolder = wrappedRendering.buildView(
+      initialViewEnvironment = initialViewEnvironment,
       contextForNewView = this.context,
       container = this
     )
-    view.start()
 
-    return buildDialogForView(view)
+    return buildDialogForView(viewHolder.view)
```

Comment:
Should change `buildDialogForView` to accept `ScreenViewHolder` instead.

Probably should update `OverlayDialogFactory` to work more like `ScreenViewFactory` -- make `DialogHolder` public, and eliminate the separate `update` method.

Reply:
Oh, derp, wrong class -- this is the legacy `buildDialogForView`. But these notes probably apply to the new stuff.

Reply:
Tracked in #713